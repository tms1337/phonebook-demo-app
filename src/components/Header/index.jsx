import React from "react";
import * as merge from "deepmerge";

const Header = ({ title, styles: propStyles = {} }) => {
  const styles = merge(defaultStyles, propStyles);

  return (
    <div
      style={{
        ...styles.container,
        ...styles.contentCentered
      }}
    >
      <h1>{title}</h1>
    </div>
  );
};

const defaultStyles = {
  container: {
    background: "black",
    height: "100px",
    color: "white"
  },
  contentCentered: {
    display: "flex",
    alignItems: "center",
    justifyContent: "center"
  }
};

export default Header;
