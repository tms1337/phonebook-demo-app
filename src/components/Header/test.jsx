import React from "react";
import renderer from "react-test-renderer";

import Header from ".";

describe("Header", () => {
  it("Snapshot renders", () => {
    const component = renderer.create(<Header />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
