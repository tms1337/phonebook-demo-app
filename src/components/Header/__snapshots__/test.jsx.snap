// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Snapshot renders 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "background": "black",
      "color": "white",
      "display": "flex",
      "height": "100px",
      "justifyContent": "center",
    }
  }
>
  <h1 />
</div>
`;
